install_config/configure_kerberos_for_sso.adoc

```diff
@@ -0,0 +1,174 @@
+==== Configuring an Apache proxy or Single Sign On authentication via Kerberos
+
+Configuring SSO authentication makes life much easier for users that are
+enrolled in a Windows Active Directory domain or a Linux IPA domain. Both these
+enteprise authentication systems use Kerberos and that allows a seamless
+experience where the user doesn't need to use additional passwords to
+authenticate to systems and services.
+
+The key challenge in getting this setup to work properly is correct
+configuration of domains names. Kerberos realies on correct DNS resolution and
+fully qualified domain names to resolve how to acquire the proper token to
+authenticate to a network service. Naming errors are the leading cause of
+unsuccessful deployments.
+
+===== Installation prerequisites
+
+First establish the (public) DNS name that you are going to assign to the
+Apache Authentication Proxy. This DNS name will be used in your domain
+controller (AD, IPA, or other Kerberos Distribtuon Center) to create a new
+Kerberos Principal and Keys associated with the proxy. These keys will be
+then stored into a keytab that will be used by Apache to authenticate users.
+No other name or alias will work, so make sure to always reference and use the
+fully qualified DNS name you choose for this proxy and all relevant cluster
+configurations.
+
+Obtain a keytab for the HTTP Service Principal Name (SPN) associated to the
+chosen DNS name from your KDC (AD, IPA, other). The SPN to be created is a
+two component principal where the first component is the string `HTTP` (all
+capitals) and the second part is the fully qualified host name (based on the
+chosen DNS name for the proxy). A SPN for this service is normally represented
+in strong form like `HTTP/hostname@REALM` where the realm is generally the
+uppercase name of the domain controlled by your KDC.
+
+Assuming a domain named krb.example.com where the KDC resides we choose a DNS
+name like auth-proxy.krb.example.com. The resulting SPN we need to create in
+the kerberos realm is: `HTTP/auth-proxy.krb.example.com@KRB.EXAMPLE.COM`
+
+Once the SPN is created assign a random key to it and extract a keytab, export
+it to a file and securely transfer it to the proxy server.
+[IMPORTANT]
+====
+The keytab contains the secret key of the service and must be transfered
+securely to the target system without leving copies in files accessible by
+other users. Access to those keys allows a complete authentication bypass
+and jeopardizes the security of your installation.
+====
+
+We assume the cluster uses the public name: `cluster.krb.example.com`
+
+We also need a TLS certificate for serving content for the chosen DNS
+name. The certificate must have a SAN for the name: `auth-proxy.krb.example.com`
+or a wildcard name for the whole domain.
+
+Finally we also need a 'client' certificate that the proxy uses to proxy to the
```

Comment:
Do we have the command example for generating the cert?

Reply:
Not really, it depends on what CA you are using.